test/e2e/serial/tests/configuration.go

```diff
@@ -356,6 +360,166 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
+			fxt.IsRebootTest = true
+			nroOperObj := &nropv1.NUMAResourcesOperator{}
+			nroKey := objects.NROObjectKey()
+			Expect(fxt.Client.Get(ctx, nroKey, nroOperObj)).ToNot(HaveOccurred())
+			initialNroOperObj := nroOperObj.DeepCopy()
+
+			mcps := &mcov1.MachineConfigPoolList{}
+			Expect(fxt.Client.List(ctx, mcps)).ToNot(HaveOccurred())
+
+			if len(mcps.Items) < 2 {
+				e2efixture.Skip(fxt, "the test requires at least 2 MCPs")
+			}
+
+			nropMcps, err := nropmcp.GetListByNodeGroupsV1(context.TODO(), e2eclient.Client, nroOperObj.Spec.NodeGroups)
+			Expect(err).ToNot(HaveOccurred())
+
+			if len(nropMcps) > 1 {
+				e2efixture.Skip(fxt, "the test supports single node group")
+			}
+
+			var testMcp mcov1.MachineConfigPool
+			initialMcp := nropMcps[0]
+			for _, mcp := range mcps.Items {
+				if mcp.Name != initialMcp.Name {
+					testMcp = mcp
+				}
+			}
+
+			customPolicySupportEnabled := isCustomPolicySupportEnabled(nroOperObj)
+
+			By(fmt.Sprintf("modifying the NUMAResourcesOperator nodeGroups field to match different mcp: %q labels %q", testMcp.Name, testMcp.Labels))
+			Eventually(func(g Gomega) {
+				// we need that for the current ResourceVersion
+				err := fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(initialNroOperObj), nroOperObj)
+				g.Expect(err).ToNot(HaveOccurred())
+
+				// as filtered at the beginning of the test, it supports single node group
+				nroOperObj.Spec.NodeGroups[0].MachineConfigPoolSelector.MatchLabels = testMcp.Labels
+				err = fxt.Client.Update(context.TODO(), nroOperObj)
+				g.Expect(err).ToNot(HaveOccurred())
+			}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+
+			defer func() {
+				By("CLEANUP: reverting the changes under the NUMAResourcesOperator object")
+				nroOperObj := &nropv1.NUMAResourcesOperator{}
+				Eventually(func(g Gomega) {
+					// we need that for the current ResourceVersion
+					err := fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(initialNroOperObj), nroOperObj)
+					g.Expect(err).ToNot(HaveOccurred())
+
+					nroOperObj.Spec = initialNroOperObj.Spec
+					err = fxt.Client.Update(context.TODO(), nroOperObj)
+					g.Expect(err).ToNot(HaveOccurred())
+				}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+
+				if customPolicySupportEnabled {
+					By(fmt.Sprintf("waiting for mcp %q to start updating", initialMcp.Name))
+					Expect(deploy.WaitForMCPsCondition(fxt.Client, context.TODO(), []*machineconfigv1.MachineConfigPool{initialMcp}, machineconfigv1.MachineConfigPoolUpdating)).To(Succeed())
+					By(fmt.Sprintf("waiting for mcp %q to complete update", initialMcp.Name))
+					Expect(deploy.WaitForMCPsCondition(fxt.Client, context.TODO(), []*machineconfigv1.MachineConfigPool{initialMcp}, machineconfigv1.MachineConfigPoolUpdated)).To(Succeed())
```

Comment:
this proven to be fragile, it seems better to wait for the MCP to become Updated, but with a different `ResourceVersion`.

Reply:
indeed: AFAIK the safest way to wait is to get BOTH `resourceVersion` > current and state == Updated